### PR TITLE
Mobile design - menu

### PR DESCRIPTION
**Pull request by tenisakb**
burger2

burger1

### CODE REVIEW

File: src/components/shared/header.tsx [hunk 1]
```diff
@@ -6,16 +6,23 @@ import "../root.css";
...
   HEADER_SCREEEN_CONTAINER,
   HEADER_TEAM_PAGE
 } from "../../config";
 
 const ICON_LOGO = require("../../assets/images/logo.svg");
+const ICON_HAMBURGER = require("../../assets/images/hamburger.svg");
+const ICON_CLOSE = require("../../assets/images/close.svg");
 
-const Container = styled.div`
+const Container = styled.div<{ menuOpen: number }>`
```
Comment (EdmundMai) [inline src/components/shared/header.tsx]: boolean

File: src/components/shared/header.tsx [hunk 2]
```diff
@@ -29,9 +36,43 @@ const LogoContainer = styled.div`
...
+    display: none;
+  }
+`;
+
+const MobileNavigationContainer = styled.div`
+  @media only screen and (min-width: ${MOBILE_WIDTH}px) {
+    display: none;
+  }
+  @media only screen and (max-width: ${MOBILE_WIDTH}px) {
+    a {
```
Comment (EdmundMai) [inline src/components/shared/header.tsx]: just make a styled component for this instead of nested css

File: src/components/shared/header.tsx [hunk 3]
```diff
@@ -29,9 +36,43 @@ const LogoContainer = styled.div`
...
+  @media only screen and (max-width: ${MOBILE_WIDTH}px) {
+    display: none;
+  }
+`;
+
+const MobileNavigationContainer = styled.div`
+  @media only screen and (min-width: ${MOBILE_WIDTH}px) {
+    display: none;
+  }
+  @media only screen and (max-width: ${MOBILE_WIDTH}px) {
```
Comment (EdmundMai) [inline src/components/shared/header.tsx]: don't think you need this media query if you are doing mobile first development. `display: block`

File: src/components/shared/header.tsx [hunk 4]
```diff
@@ -108,23 +163,76 @@ const renderLinks = (type, history) => {
...
+}
+
+class Header extends React.Component<HeaderProps, HeaderState> {
+  constructor (props) {
+    super(props);
+    this.state = {
+      menuOpen: false
+    };
+  }
+
```
Comment (EdmundMai) [inline src/components/shared/header.tsx]: we use a babel plugin for this, so you can just do this: https://github.com/opacity/webinterface/blob/1bbd1ee8c476264bb576063f40a6d945b2bb60b6/src/components/signup/record-storage-handle-slide.tsx#L159

```
state = {
  menuOpen: false
}

```

File: src/components/shared/header.tsx [hunk 5]
```diff
@@ -108,23 +163,76 @@ const renderLinks = (type, history) => {
...
+            {!menuOpen ? (
+              <HamburgerIcon
+                src={ICON_HAMBURGER}
+                alt="navigation"
+                onClick={() => {
+                  this.hamburgerClick();
+                }}
+              />
+            ) : null}
+            {menuOpen ? (
```
Comment (EdmundMai) [inline src/components/shared/header.tsx]: `menuOpen && <MobileNavigationContainer>...</MobileNavigationContainer>`

and remove the `: null`

File: src/components/shared/header.tsx [hunk 6]
```diff
@@ -108,23 +163,76 @@ const renderLinks = (type, history) => {
...
+            <LogoContainer>
+              <Link
+                title="Opacity Storage's Logo"
+                onClick={() => history.push("/")}
+              >
+                <Logo src={ICON_LOGO} alt="logo" />
+                <CompanyName>Opacity</CompanyName>
+              </Link>
+            </LogoContainer>
+            {!menuOpen ? (
```
Comment (EdmundMai) [inline src/components/shared/header.tsx]: `{ !menuOpen && <Hamburger/> }`

File: src/components/shared/header.tsx (hunk 1 above)
Comment (tenisakb) [reply src/components/shared/header.tsx]: fixed

File: src/components/shared/header.tsx (hunk 3 above)
Comment (tenisakb) [reply src/components/shared/header.tsx]: fixed

File: src/components/shared/header.tsx (hunk 2 above)
Comment (tenisakb) [reply src/components/shared/header.tsx]: fixed

File: src/components/shared/header.tsx (hunk 4 above)
Comment (tenisakb) [reply src/components/shared/header.tsx]: fixed

File: src/components/shared/header.tsx (hunk 5 above)
Comment (tenisakb) [reply src/components/shared/header.tsx]: fixed

File: src/components/shared/header.tsx (hunk 6 above)
Comment (tenisakb) [reply src/components/shared/header.tsx]: fixed

**Pull request merged**